### PR TITLE
Deploy-chain

### PR DESCRIPTION
**Pull request by FritzHoing**
This MR implements a deploy chain for the following images using semantic release:
- shogun-gis-client-docs (latest)
- shogun-gis-client-docs (version)
- shogun-admin-client-docs (latest)
- shogun-admin-client-docs (version)

### CODE REVIEW

File: .github/workflows/gh-pages.yml [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+  create_and_push_image:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout sources 🔰
+      uses: actions/checkout@v4
+
+    - name: Setup Node.js 🧮
+      uses: actions/setup-node@v4
+      with:
+        node-version: 20
```
Comment (dnlkoch) [inline .github/workflows/gh-pages.yml:53]: In the job above we make use of node in version 20 and an older action version. We should check if we can harmonize this (and potentially others).

File: .github/workflows/gh-pages.yml [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+      uses: cycjimmy/semantic-release-action@v4
+      id: semantic
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
+      with:
+        semantic_version: 23
+
+    - name: Build artifacts 🏗️
+      run: npm run build
```
Comment (dnlkoch) [inline .github/workflows/gh-pages.yml]: Is this needed?

File: .github/workflows/gh-pages.yml [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+    
+    - name: Login to Nexus ⌨️
+      uses: docker/login-action@v1
+      with:
+        registry: ${{ env.DOCKER_REGISTRY }}
+        username: ${{ secrets.NEXUS_USERNAME }}
+        password: ${{ secrets.NEXUS_PASSWORD }}
+
+    - name: Build docker gis-client image (latest) 🏗️
```
Comment (dnlkoch) [inline .github/workflows/gh-pages.yml]:
```suggestion
    - name: Build shogun-gis-client-docs docker image (latest) 🏗️
```

File: .github/workflows/gh-pages.yml [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+      uses: docker/build-push-action@v5
+      with:
+        context: .
+        file: Dockerfile.gis.client
+        tags: |
+          ${{ env.DOCKER_REGISTRY }}/shogun-gis-client-docs:latest
+        load: true
+
+    - name: Build docker gis-client image (version)
```
Comment (dnlkoch) [inline .github/workflows/gh-pages.yml]:
```suggestion
    - name: Build shogun-gis-client-docs docker image (version) 🏗️
```

File: .github/workflows/gh-pages.yml [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+      if: steps.semantic.outputs.new_release_published == 'true'
+      uses: docker/build-push-action@v5
+      with:
+        context: .
+        file: Dockerfile.gis.client
+        tags: |
+          ${{ env.DOCKER_REGISTRY }}/shogun-gis-client-docs:${{ steps.semantic.outputs.new_release_version }}
+        load: true
+
+    - name: Push docker gis-client image to Nexus (latest) 📠
```
Comment (dnlkoch) [inline .github/workflows/gh-pages.yml]:
```suggestion
    - name: Push shogun-gis-client-docs docker image to Nexus (latest) 📠
```

File: .github/workflows/gh-pages.yml [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+      run: |
+        docker push ${{ env.DOCKER_REGISTRY }}/shogun-gis-client-docs:latest
+
+    - name: Push docker gis-client image to Nexus (version)
```
Comment (dnlkoch) [inline .github/workflows/gh-pages.yml]:
```suggestion
    - name: Push shogun-gis-client-docs docker image to Nexus (version) 📠
```

File: .github/workflows/gh-pages.yml [hunk 7]
```diff
@@ -0,0 +1,140 @@
...
+      if: steps.semantic.outputs.new_release_published == 'true'
+      run: |
+        docker push ${{ env.DOCKER_REGISTRY }}/shogun-gis-client-docs:${{ steps.semantic.outputs.new_release_version }}
+
+    - name: Build docker admin-client image (latest) 🏗️
```
Comment (dnlkoch) [inline .github/workflows/gh-pages.yml]:
```suggestion
    - name: Build shogun-admin-client-docs docker image (latest) 🏗️
```

File: .github/workflows/gh-pages.yml [hunk 8]
```diff
@@ -0,0 +1,140 @@
...
+      uses: docker/build-push-action@v5
+      with:
+        context: .
+        file: Dockerfile.admin.client
+        tags: |
+          ${{ env.DOCKER_REGISTRY }}/shogun-admin-client-docs:latest
+        load: true
+
+    - name: Build docker admin-client image (version)
```
Comment (dnlkoch) [inline .github/workflows/gh-pages.yml]:
```suggestion
    - name: Build shogun-admin-client-docs docker image (version) 🏗️
```

File: .github/workflows/gh-pages.yml [hunk 9]
```diff
@@ -0,0 +1,140 @@
...
+      if: steps.semantic.outputs.new_release_published == 'true'
+      uses: docker/build-push-action@v5
+      with:
+        context: .
+        file: Dockerfile.admin.client
+        tags: |
+          ${{ env.DOCKER_REGISTRY }}/shogun-admin-client-docs:${{ steps.semantic.outputs.new_release_version }}
+        load: true
+
+    - name: Push docker admin-client image to Nexus (latest) 📠
```
Comment (dnlkoch) [inline .github/workflows/gh-pages.yml]:
```suggestion
    - name: Push shogun-admin-client-docs docker image to Nexus (latest) 📠
```

File: .github/workflows/gh-pages.yml [hunk 10]
```diff
@@ -0,0 +1,140 @@
...
+      run: |
+        docker push ${{ env.DOCKER_REGISTRY }}/shogun-admin-client-docs:latest
+    
+    - name: Push docker admin-client image to Nexus (version)
```
Comment (dnlkoch) [inline .github/workflows/gh-pages.yml]:
```suggestion
    - name: Push shogun-admin-client-docs docker image to Nexus (version) 📠
```

File: .releaserc.json [hunk 11]
```diff
@@ -0,0 +1,63 @@
...
+          {"type": "test", "release": false}
+        ]
+      }
+    ],
+    [
+      "@semantic-release/release-notes-generator",
+      {
+        "preset": "conventionalcommits",
+        "presetConfig": {
+          "header": "Changelog of shogun-admin",
```
Comment (dnlkoch) [inline .releaserc.json]:
```suggestion
          "header": "Changelog of shogun-docs",
```

File: .github/workflows/gh-pages.yml [hunk 12]
```diff
@@ -0,0 +1,140 @@
...
+
+    - name: Install dependencies ⏬
+      run: npm ci
+
+    - name: Semantic Release 🚀
+      uses: cycjimmy/semantic-release-action@v4
+      id: semantic
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
```
Comment (dnlkoch) [inline .github/workflows/gh-pages.yml]: Since `npmPublish` is deactivated I assume this is not needed.

**Comment by FritzHoing** [comment]
Thank you very much for the thorough review @dnlkoch ! All suggestions should be implemented now.

File: .github/workflows/gh-pages.yml (hunk 1 above)
Comment (FritzHoing) [reply .github/workflows/gh-pages.yml:53]: Thank you for the good hint! I harmonized the node versions and tested the pipeline

File: Dockerfile.admin.client [hunk 13]
```diff
@@ -1,4 +1,4 @@
-FROM node:16.15.1-alpine as build
+FROM node:18.18.2-alpine3.17 AS build
```
Comment (dnlkoch) [inline Dockerfile.admin.client]: To be in sync with the node versions used in the jobs, we might want to update to version 20 herein as well.

File: Dockerfile.gis.client [hunk 14]
```diff
@@ -0,0 +1,21 @@
+FROM node:18.18.2-alpine3.17 AS build
```
Comment (dnlkoch) [inline Dockerfile.gis.client]: See above.

**Review by dnlkoch — APPROVED** [review]

File: Dockerfile.admin.client (hunk 13 above)
Comment (FritzHoing) [reply Dockerfile.admin.client]: Thank you! I am using `20.12.1-alpine3.19` now

**Pull request merged**